ppb/systems/renderer.py

```diff
@@ -152,15 +156,23 @@ def __exit__(self, *exc):
     def on_idle(self, idle_event: events.Idle, signal):
         t = monotonic()
         if t >= self.target_clock:
-            self.pre_render_updates(idle_event.scene)
             signal(events.PreRender())
             signal(events.Render())
             self.target_clock = t + self.target_frame_length
 
-    def pre_render_updates(self, scene):
-        camera = scene.main_camera
-        camera.viewport_width, camera.viewport_height = self.resolution
-        self.pixel_ratio = camera.pixel_ratio
+    def on_scene_started(self, scene_started, signal):
+        scene = scene_started.scene
+        camera_class = getattr(scene_started.scene, "camera_class", Camera)
+        # For future: This is basically the pattern we'd use to define
+        # multiple cameras. We'd just need to have the scene tell us the
+        # regions they should render to.
+        camera = camera_class(self, self.target_camera_width, self.resolution)
+        scene.main_camera = camera
+        self.scene_cameras[scene] = [camera]
```

Comment:
Why the extra ref?

Reply:
Primarily to have handles to adjust resolution when we finally start supporting resizeable windows.